docs/design/automatic-updates.md

```diff
@@ -0,0 +1,38 @@
+
+# Storage Node automatic updates DesignDocument
+
+## Overview
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date. If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.
+
+## Goals
+
+Create an automatic update system into the Storage Node software so that node operators do not have to worry about updating their nodes.
+Keep the Storage Nodes on the network up to date with the latest software to improve stability and performance. 
+Concerns
+
+A majority or substantial amount of nodes update their software at the same time causing instability in the network or temporary file lose.
+We must ensure the automatic update system we build into the Storage nodes works on ALL OS we support.
+
+## Scenarios
+
+
+## Design Overview
+
+When starting a storj node:
+* Main process starts an updater service which starts a (12 hour?) interval loop
+* Updater service spawns a storj node process 
+* Updater service Interval checks the current storj node version and compares with the version server ("https://version.alpha.storj.io/") 
+* Updater service Downloads the Binary for the minimum version returned from the version server
```

Comment:
Where does it download the binary from?  Do we have a process in place for this yet?

Reply:
Could the updater service just pull the tag from github, and install using `go install ./cmd/storagenode`?

Reply:
This would require installing `go` on the computer.

Reply:
So what is the advantage of writing our own auto update system rather than using watchtower?

Reply:
watchtower requires docker.

Reply:
download the binary from github releases by ag

Reply:
If we are downloading from github, could we get an issue with rate limiting?

Reply:
It would be no different than all the users going to the github release page and clicking the download button.

Reply:
The difference is that we are doing it automated so we could get a flood of the requests in a small timeframe. But, is there rate-limiting per repository?

Reply:
https://help.github.com/en/articles/about-releases#limitations-on-binary-files
"We don't limit the total size of your binary release files, nor the bandwidth used to deliver them. However, each individual file must be under 2 GB in size."

Reply:
Side note, you can pull a list releases / files from Github using GraphQL...